pi-manage

```diff
@@ -297,25 +298,26 @@ def create(directory="/var/lib/privacyidea/backup/",
     backup_file = "%s/%s-%s.tgz" % (directory, BASE_NAME, DATE)
 
     sqluri = app.config.get("SQLALCHEMY_DATABASE_URI")
-    sqltype = sqluri.split(":")[0]
+    parsed_sqluri = urlparse(app.config.get("SQLALCHEMY_DATABASE_URI"))
+    sqltype = parsed_sqluri.scheme
+
     if sqltype == "sqlite":
         productive_file = sqluri[len("sqlite:///"):]
         print("Backup SQLite %s" % productive_file)
         sqlfile = "%s/db-%s.sqlite" % (directory, DATE)
         call(["cp", productive_file, sqlfile])
     elif sqltype in MYSQL_DIALECTS:
-        m = re.match(r".*mysql://(.*):(.*)@(.*)/(\w*)\??(.*)", sqluri)
-        username = m.groups()[0]
-        password = m.groups()[1]
-        datahost = m.groups()[2]
-        database = m.groups()[3]
+        username = parsed_sqluri.username
+        password = parsed_sqluri.password
+        hostname = parsed_sqluri.hostname
+        database = parsed_sqluri.path[1:]
         # We strip parameters, but we do not use them
-        _parameters = m.groups()[4]
+        _parameters = parsed_sqluri.params
```

Comment:
I think we can remove these 2 lines if they are not used

Reply:
fixed in 71167a5072fe3f5e7ef36903d2e68649ab66a5a7.